cartesi-rollups/tutorials/erc-20-token-wallet.md

```diff
@@ -0,0 +1,391 @@
+      throw new Error("ERC20 balance is undefined");
+    }
+
+    await sendReport({ payload: toHex(erc20balance) });
+  } catch (error) {
+    console.error("Error processing inspect payload:", error);
+  }
+}
+
+const sendNotice = async (payload: Payload) => {
+  await fetch(`${rollupServer}/notice`, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify(payload),
+  });
+};
+
+const sendVoucher = async (payload: Voucher) => {
+  await fetch(`${rollupServer}/voucher`, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify({ payload }),
+  });
+};
+
+const sendReport = async (payload: Report) => {
+  await fetch(`${rollupServer}/report`, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify({ payload }),
+  });
+};
+
+const main = async () => {
+  const { POST } = createClient<paths>({ baseUrl: rollupServer });
+  let status: RequestHandlerResult = "accept";
+  while (true) {
+    const { response } = await POST("/finish", {
+      body: { status },
+      parseAs: "text",
+    });
+
+    if (response.status === 200) {
+      const data = (await response.json()) as RollupsRequest;
+      switch (data.request_type) {
+        case "advance_state":
+          status = await handleAdvance(data.data as AdvanceRequestData);
+          break;
+        case "inspect_state":
+          await handleInspect(data.data as InspectRequestData);
+          break;
+      }
+    } else if (response.status === 202) {
+      console.log(await response.text());
+    }
+  }
+};
+
+main().catch((e) => {
+  console.log(e);
+  process.exit(1);
+});
+```
+
+
+Here is a breakdown of the wallet functionality:
+- We handle deposits when the sender is the `ERC20Portal`.
+- For other senders, we parse the payload to determine the operation (`transfer` or `withdraw`).
+- For `transfers`, we call `wallet.transferERC20` with the parsed parameters.
+- For `withdrawals`, we call `wallet.withdrawERC20` with the parsed parameters.
+- We created helper functions to `sendNotice` for deposits and transfers, `sendReport` for balance checks and `sendVoucher` for withdrawals.
+
+
+#### Deposits
+To deposit ERC20 tokens, use the `cartesi send erc20` command and follow the prompts.
+
+#### Balance checks(used in Inspect requests)
+
+To inspect balance, make an HTTP call to:
+
+```
+http://localhost:8080/inspect/{address}/{tokenAddress}
+```
+
+
+#### Transfers and Withdrawals
+
+Use the `cartesi send generic` command and follow the prompts. Here are sample payloads:
+
+1. For transfers:
+
+	```js
+	{"operation":"transfer","erc20":"0xTokenAddress","from":"0xFromAddress","to":"0xToAddress","amount":"1000000000000000000"}
+	```
+
+2. For withdrawals:
+
+	```js
+	{"operation":"withdraw","erc20":"0xTokenAddress","from":"0xFromAddress","amount":"1000000000000000000"}
+	```
+
+:::note community tools
```

Comment:
I'm inclined towards adding this at the top so that one can directly use the framework if they don't want to go into basics.